new-log-viewer/src/services/decoders/JsonlDecoder/utils.ts

```diff
@@ -0,0 +1,100 @@
+import dayjs, {Dayjs} from "dayjs";
+import utc from "dayjs/plugin/utc";
+
+import {
+    JsonObject,
+    JsonValue,
+} from "../../../typings/js";
+import {
+    INVALID_TIMESTAMP_VALUE,
+    LOG_LEVEL,
+} from "../../../typings/logs";
+
+
+// eslint-disable-next-line import/no-named-as-default-member
+dayjs.extend(utc);
+
+/**
+ * A log event parsed from a JSON log.
+ */
+interface JsonLogEvent {
+    timestamp: Dayjs,
+    level: LOG_LEVEL,
+    fields: JsonObject
+}
+
+/**
+ * Narrow JSON value to JSON object if compatible.
+ * Reference: https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates
+ *
+ * @param fields
+ * @return Whether type is JsonObject.
+ */
+const isJsonObject = (fields: JsonValue): fields is JsonObject => {
+    return "object" === typeof fields;
+};
+
+/**
+ * Converts JSON log level field into a log level value.
+ *
+ * @param logLevelField
+ * @return Integer for log level.
+ */
+const convertToLogLevelValue = (logLevelField: JsonValue | undefined): number => {
+    let logLevelValue = LOG_LEVEL.NONE;
+
+    if ("undefined" === typeof logLevelField) {
+        return logLevelValue;
+    }
+
+    const logLevelName = "object" === typeof logLevelField ?
+        JSON.stringify(logLevelField) :
+        String(logLevelField);
+
+    if (logLevelName.toUpperCase() in LOG_LEVEL) {
+        logLevelValue = LOG_LEVEL[logLevelName.toUpperCase() as keyof typeof LOG_LEVEL];
+    }
+
+    return logLevelValue;
+};
+
+/**
+ * Converts JSON timestamp field into a dayjs timestamp.
+ *
+ * @param timestampField
+ * @return The timestamp or `INVALID_TIMESTAMP_VALUE` if:
+ * 1. the timestamp key doesn't exist in the log.
+ * 2. the timestamp's value is an unsupported type.
+ * 3. the timestamp's value is not a valid dayjs timestamp.
```

Comment:
```suggestion
 * @return The field as a dayjs timestamp or `dayjs.utc(INVALID_TIMESTAMP_VALUE)` if:
 * - the timestamp key doesn't exist in the log.
 * - the timestamp's value is an unsupported type.
 * - the timestamp's value is not a valid dayjs timestamp.
```

I try to reserve numbering for things that actually must be sequential.